engine/gamesys/proto/gamesys/atlas_ddf.proto

```diff
@@ -13,6 +13,10 @@ message AtlasImage
 {
     required string image                           = 1 [(resource) = true];
     optional SpriteTrimmingMode sprite_trim_mode    = 2 [default = SPRITE_TRIM_MODE_OFF];
+    optional int32 source_position_x                = 3 [default = 0];
+    optional int32 source_position_y                = 4 [default = 0];
+    optional uint32 source_width                    = 5 [default = 0];
+    optional uint32 source_height                   = 6 [default = 0];
```

Comment:
For any feature this large/important, we want to have a design phase at the beginning, in order to make sure we consider as many aspects as possible.

Currently, I feel that the file format change may not be what we want.
For instance, it looks like the texture packer result is extracted, into separate images(?) and then assembled again in our file format?
That's certainly something we need to think about.
E.g. an idea we've had is to store the actual packing info in the atlas file.

Please start a design document (e.g. google docs) where we can have discussions about the feature.

Reply:
This feature was designed and developed for a company and is now being used there internally, they were kind enough to share it with the community. Hence there was no public design document since it's basically an internal feature.

> For instance, it looks like the texture packer result is extracted, into separate images(?) and then assembled again in our file format?

TexturePacker exporter generates a ready atlas file for Defold, unfortunately it's not capable of exporting individual sprite images for Defold. Therefore I've created the spritesheet splitting script that just splits the TexturePacker's spritesheet into separate files. The generated JSON file is used just for that, the script does not modify the atlas file and the JSON file is deleted after the splitting.

> E.g. an idea we've had is to store the actual packing info in the atlas file.

Sorry, not understanding this. What other packing info? In this feature the packing info is already stored in the atlas file alone. The json file is not needed. These four new fields of images inside atlas are all that is needed for that feature.

I've created the design document file as you requested. https://docs.google.com/document/d/1fm5zHKLUZzMVYmLE2E7aFpb_2vBDxGB7qWhPHooL3Uw/edit?usp=sharing

We were not sure about Defold's roadmap in regard to atlas or texture improvements that's why we designed and developed this feature in this manner to meet the current project needs. We were designing the feature taking into account current Defold architecture with its pros and cons.
You can check the design file and we can discuss it and I can help with the Java side of things, but I would rather not touch the Clojure code, it's way to hard for me. The most important part of the feature's code is in [com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/textureset/TextureSetGenerator.java](https://github.com/defold/defold/pull/7424#diff-66aa1f9e9cb6b575e7454c7a8812104ae2b15f2e1b26d565b10510ad54d70393) anyway.